### ISSUE TITLE
Custom syntax rules

### ISSUE THREAD

**Issue by lete114** [markedjs/marked#2061]
I have a need for custom syntax rules such as
```js
import marked from "marked";

marked(":[emoji]");
// output: <img src="https://xxxx.com/img/emoji.png" alt="emoji" />

// demo
// marked(":[emoji-demo]");
// output: <img src="https://xxxx.com/img/emoji-demo.png" alt="emoji-demo" />
```
Are there any good ideas? I've seen a lot of examples online of directly modifying the source code, but this is not what I want (too complicated).

**Comment by UziTech** [comment]
We are working on a PR to add custom rules #2043.

**Comment by UziTech** [comment]
This Should be available in v2.1.0

see Custom Extensions section in [docs](https://marked.js.org/using_pro)
